### PR TITLE
Docs: Update Loki documentation

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
Updates documentation for Loki data source

### CODE REVIEW

**Comment by osg-grafana** [comment]
@achatterjee-grafana, can you add the Docs Squad as a code owner?

**Comment by achatterjee-grafana** [comment]
@osg-grafana done. I am working with Ivana on this PR, and also David has a section to his name, that is why we got added. (Clarification).

File: docs/sources/datasources/loki.md [hunk 1]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
...
-- `{instance=~"kafka-[23]",name="kafka"} != "kafka.server:type=ReplicaManager"`
-
-Filter operators can be chained and will sequentially filter down the expression. The resulting log lines will satisfy every filter.
+### Log Queries
 
-**Example**
+Querying and displaying log data from Loki is available via [Explore]({{< relref "../explore" >}}), and with the [logs panel]({{< relref "../panels/visualizations/logs-panel.md" >}}) in dashboards. Select the Loki data source, and then enter a [LogQL](https://grafana.com/docs/loki/latest/logql/#log-queries) query to display your logs.
 
-`{job="mysql"} |= "error" != "timeout"`
+A log query consists of two parts: **log stream selector**, and a **log pipeline**. For performance reasons you need to start by choosing a log stream by selecting a log label.
```
Comment (oddlittlebird) [inline docs/sources/datasources/loki.md:77]: Why are the terms bold? If they are not on the UI, they should not be bold. Apply italic if you want to emphasize them.

File: docs/sources/datasources/loki.md [hunk 2]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
...
 
-The following filter types are currently supported:
+### Log Context
```
Comment (oddlittlebird) [inline docs/sources/datasources/loki.md:79]:
```suggestion
### Log context
```

File: docs/sources/datasources/loki.md [hunk 3]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
...
 
-- `|=` line contains string.
-- `!=` line doesn't contain string.
-- `|~` line matches regular expression.
-- `!~` line does not match regular expression.
+When using a search expression as detailed above, you now have the ability to retrieve the context surrounding your filtered results.
```
Comment (oddlittlebird) [inline docs/sources/datasources/loki.md:81]:
```suggestion
When using a search expression as detailed above, you can retrieve the context surrounding your filtered results.
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
Great changes! Here are some of my suggestions.

File: docs/sources/datasources/loki.md [hunk 4]
```diff
@@ -6,31 +6,25 @@ aliases = ["/docs/grafana/latest/features/datasources/loki"]
...
-> In Grafana v5.3.x and v5.4.x. you need to enable Explore manually.
-> Viewing Loki data in dashboard panels is supported in Grafana v6.4+.
+Grafana ships with built-in support for Loki, an open source log aggregation system by Grafana Labs. Add it as a data source and you are ready to build dashboards or query your log data in [Explore]({{< relref "../explore" >}}).
 
-Grafana ships with built-in support for Loki, Grafana's log aggregation system.
-Just add it as a data source and you are ready to query your log data in [Explore]({{< relref "../explore" >}}).
+This topic explains options, variables, querying, and other options specific to this data source. Refer to [Add a data source]({{< relref "add-a-data-source.md" >}}) for instructions on how to add a data source to Grafana. Only users with the organization admin role can add data sources.
 
-## Adding the data source
+> **Note:** If you have issues with getting this data source to work and Grafana is giving you undescriptive errors, then check your log file (try looking in /var/log/grafana/grafana.log).
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:15]:
```suggestion
> **Note:** To troubleshoot configuration and other issues, check the log file (/var/log/grafana/grafana.log).
```

File: docs/sources/datasources/loki.md [hunk 5]
```diff
@@ -6,31 +6,25 @@ aliases = ["/docs/grafana/latest/features/datasources/loki"]
...
+| Name                  | Description                                                                                                                                   |
+| --------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Name`                | The data source name. This is how you refer to the data source in panels, queries, and Explore.                                               |
+| `Default`             | Default data source means that it will be pre-selected for new panels.                                                                        |
+| `URL`                 | The URL of the Loki instance, e.g., `http://localhost:3100`                                                                                   |
+| `Whitelisted Cookies` | Grafana Proxy deletes forwarded cookies by default. Specify cookies by name that should be forwarded to the data source.                      |
+| `Maximum lines`       | Upper limit for number of log lines returned by Loki (default is 1000). Decrease if your browser is sluggish when displaying logs in Explore. |
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:27]:
```suggestion
| `Maximum lines`       | Upper limit for the number of log lines returned by Loki (default is 1000). Lower this limit if your browser is sluggish when displaying logs in Explore. |
```

File: docs/sources/datasources/loki.md [hunk 6]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
...
 
-Examples:
+## Querying with Loki
 
-- `{name=~"mysql.+"}`
-- `{name!~"mysql.+"}`
+There are two types of LogQL queries:
 
-The [same rules that apply for Prometheus Label Selectors](https://prometheus.io/docs/prometheus/latest/querying/basics/#instant-vector-selectors) apply for Loki Log Stream Selectors.
+- **Log queries** that return the contents of log lines.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:70]:
```suggestion
- **Log queries:** returns the contents of log lines.
```

File: docs/sources/datasources/loki.md [hunk 7]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
...
+- **Metric queries** that extend log queries and calculate sample values based on the content of logs from a log query.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:71]:
```suggestion
- **Metric queries:** extends log queries and calculate sample values based on the content of logs from a log query.
```

File: docs/sources/datasources/loki.md (hunk 1 above)
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:77]:
```suggestion
A log query consists of two parts: **log stream selector**, and a **log pipeline**. For performance reasons begin by choosing a log stream by selecting a log label.
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added some comments. Will add the rest ASAP.

File: docs/sources/datasources/loki.md [hunk 8]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
...
-- `|=` line contains string.
-- `!=` line doesn't contain string.
-- `|~` line matches regular expression.
-- `!~` line does not match regular expression.
+When using a search expression as detailed above, you now have the ability to retrieve the context surrounding your filtered results.
+By clicking the `Show Context` link on the filtered rows, you'll be able to investigate the log messages that came before and after the
+log message you're interested in.
 
-> **Note:** For more details about LogQL, Loki's query language, refer to the [Loki LogQL](https://grafana.com/docs/loki/latest/logql/)
+> **Note:** This feature is only available in Grafana v6.3+
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:85]: Can we remove this note? 6.3 is really old.

File: docs/sources/datasources/loki.md [hunk 9]
```diff
@@ -6,31 +6,25 @@ aliases = ["/docs/grafana/latest/features/datasources/loki"]
...
 +++
 
-# Using Loki in Grafana
+# Loki data source
 
-> BETA: Querying Loki data requires Grafana's Explore section.
-> Grafana v6.x comes with Explore enabled by default.
-> In Grafana v5.3.x and v5.4.x. you need to enable Explore manually.
-> Viewing Loki data in dashboard panels is supported in Grafana v6.4+.
+Grafana ships with built-in support for Loki, an open source log aggregation system by Grafana Labs. Add it as a data source and you are ready to build dashboards or query your log data in [Explore]({{< relref "../explore" >}}).
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:11]:
```suggestion
Grafana ships with built-in support for Loki, an open source log aggregation system by Grafana Labs. This topic explains options, variables, querying, and other options specific to this data source.

Add it as a data source and you are ready to build dashboards or query your log data in [Explore]({{< relref "../explore" >}}). Refer to [Add a data source]({{< relref "add-a-data-source.md" >}}) for instructions on how to add a data source to Grafana. Only users with the organization admin role can add data sources.
```

File: docs/sources/datasources/loki.md [hunk 10]
```diff
@@ -6,31 +6,25 @@ aliases = ["/docs/grafana/latest/features/datasources/loki"]
...
 
-Grafana ships with built-in support for Loki, Grafana's log aggregation system.
-Just add it as a data source and you are ready to query your log data in [Explore]({{< relref "../explore" >}}).
+This topic explains options, variables, querying, and other options specific to this data source. Refer to [Add a data source]({{< relref "add-a-data-source.md" >}}) for instructions on how to add a data source to Grafana. Only users with the organization admin role can add data sources.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:13]: Move content up, as shown.

File: docs/sources/datasources/loki.md [hunk 11]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
 The new field with the link shown in log details:
 {{< docs-imagebox img="/img/docs/v65/loki_derived_fields_detail.png" class="docs-image--no-shadow" caption="Screenshot of the derived field in log detail" >}}
 
-## Querying Logs
+## Loki Query Editor
 
-Querying and displaying log data from Loki is available via [Explore]({{< relref "../explore" >}}), and with the [logs panel]({{< relref "../panels/visualizations/logs-panel.md" >}}) in dashboards. Select the Loki data source, and then enter a [LogQL](https://grafana.com/docs/loki/latest/logql/) query to display your logs.
+Loki query editor can be used to create both log and metric queries.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:53]:
```suggestion
Use the Loki query editor to create both log and metric queries.
```

File: docs/sources/datasources/loki.md [hunk 12]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
...
-<br />
-
-### Log Stream Selector
-
-For the label part of the query expression, wrap it in curly braces `{}` and then use the key value syntax for selecting labels. Multiple label expressions are separated by a comma:
-
-`{app="mysql",name="mysql-backup"}`
+| Name               | Description                                                                                                                                                                                                 |
+| ------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Query expression` | Loki query expression, check out the [LogQL documentation](https://grafana.com/docs/loki/latest/logql/).                                                                                                    |
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:57]:
```suggestion
| `Query expression` | Loki query expression, refer to the [LogQL documentation](https://grafana.com/docs/loki/latest/logql/) for more information.                                                                                                    |
```

File: docs/sources/datasources/loki.md [hunk 13]
```diff
@@ -54,92 +48,65 @@ You can use a debug section to see what your fields extract and how the URL is i
...
+| `Query type`       | Choose the type of query you would like to run. An instant query queries against a single point in time. We are using "To" time from the time range. A range query queries over the selected range of time. |
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/loki.md:58]:
```suggestion
| `Query type`       | Choose the type of query to run. The instant type queries against a single point in time. We are using "To" time from the time range. The range type queries over the selected range of time. |
```

File: docs/sources/datasources/loki.md (hunk 6 above)
Comment (achatterjee-grafana) [reply docs/sources/datasources/loki.md:70]: Log queries do not need to be bold.

File: docs/sources/datasources/loki.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (achatterjee-grafana) [reply docs/sources/datasources/loki.md:71]: Metric queries do not need to be bold.

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Some more review comments. Thanks for updating the topic Ivana!

**Comment by osg-grafana** [comment]
> @osg-grafana done. I am working with Ivana on this PR, and also David has a section to his name, that is why we got added. (Clarification).

Thank you. I will keep an eye on this content to align on it when working on GEL.

File: docs/sources/datasources/loki.md (hunk 4 above)
Comment (oddlittlebird) [reply docs/sources/datasources/loki.md:15]: This path is not going to work for all users. On my Windows machine, it's in grafana/data/log/grafana.log. I don't have a var folder.

File: docs/sources/datasources/loki.md (hunk 1 above)
Comment (ivanahuckova) [reply docs/sources/datasources/loki.md]: Removed bold. I guess, they don't need to be bold.

File: docs/sources/datasources/loki.md (hunk 6 above)
Comment (ivanahuckova) [reply docs/sources/datasources/loki.md]: Removed bold. Plus this was changed to `Log queries -`

File: docs/sources/datasources/loki.md (hunk 8 above)
Comment (ivanahuckova) [reply docs/sources/datasources/loki.md:85]: removed

**Comment by ivanahuckova** [comment]
1. I have addressed all feedback.
2. Added log browser documentation: https://github.com/grafana/grafana/pull/31867/commits/1e3bb4d6924a58357602e412c1a608ebc5736ca8
3. New images added in: https://github.com/grafana/website/pull/3814

File: docs/sources/datasources/loki.md [hunk 14]
```diff
@@ -49,97 +43,73 @@ Each derived field consists of:
...
+| `Legend`           | Available only in Dashboard. Controls the name of the time series, using name or pattern. For example `{{hostname}}` is replaced with the label value for the label `hostname`.                             |
 
-Another way to add a label selector is in the table section. Click **Filter** beside a label to add the label to the query expression. This even works for multiple queries and will add the label selector to each query.
+### Log browser
 
-### Search Expression
+With Loki log browser you can easily navigate trough your list of labels and values and construct the query of your choice. Log browser has multi-step selection:
+1. Choose the labels you would like to consider for your search.
+2. Pick the values for selected labels. Log browser supports facetting and therefore it shows you only possible label combinations. 
+3. Choose the type of query - logs query or rate metrics query. Additionaly, you can also validate selector.
```
Comment (ivanahuckova) [inline docs/sources/datasources/loki.md:67]:
```suggestion
3. Choose the type of query - logs query or rate metrics query. Additionally, you can also validate selector.
```

**Review by oddlittlebird — APPROVED** [review]
Great work, Ivana!

**Review by achatterjee-grafana — APPROVED** [review]
Great work!

**Pull request merged**